### PR TITLE
destination: Check port bounds

### PR DESCRIPTION
**Pull request by olix0r**
CodeQL analysis flags that our use of `strconv.Atoi` is potentially
incorrect. More information [here][1].

This change addresses this by explicitly checking the bounds of the port
integer before casting it from `int` to `uint32`.

[1]: https://codeql.github.com/codeql-query-help/go/go-incorrect-integer-conversion/

### CODE REVIEW

**Review by kleimkuhler — COMMENTED** [review]
I know it's just draft right now but we should update `validateRangeSlice` in `range_slice.go` as well.

**Comment by olix0r** [comment]
> I know it's just draft right now but we should update `validateRangeSlice` in `range_slice.go` as well.

Turns out we already are doing this validation

https://github.com/linkerd/linkerd2/blob/2cd2ecfa30b248c30bd17a46c246b2db048d8500/cli/cmd/range_slice.go#L40-L44

File: .github/workflows/static_checks.yml [hunk 1]
```diff
@@ -40,7 +40,14 @@ jobs:
...
         run: apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-recommends install unzip
       - name: Checkout code
+<<<<<<< HEAD
+          uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+||||||| 685b1d43
+        # actions/checkout@v2
+        uses: actions/checkout@722adc63f1aa60a57ec37892e133b1d319cae598
+=======
         uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+>>>>>>> main
```
Comment (olix0r) [inline .github/workflows/static_checks.yml:50]:
```suggestion
        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
```

**Review by mateiidavid — APPROVED** [review]
👍🏻  looks good

**Review by dadjeibaah — APPROVED** [review]
📦 🚢

**Review by alpeb — APPROVED** [review]
:+1:

**Review by adleong — APPROVED** [review]

**Pull request merged**